app/src/androidTest/java/ch/epfl/qedit/LoginActivityLanguageTest.java

```diff
@@ -0,0 +1,92 @@
+package ch.epfl.qedit;
+
+import static androidx.test.espresso.Espresso.onData;
+import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.action.ViewActions.click;
+import static androidx.test.espresso.action.ViewActions.closeSoftKeyboard;
+import static androidx.test.espresso.assertion.ViewAssertions.matches;
+import static androidx.test.espresso.matcher.RootMatchers.withDecorView;
+import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static androidx.test.espresso.matcher.ViewMatchers.withSpinnerText;
+import static androidx.test.espresso.matcher.ViewMatchers.withText;
+import static org.hamcrest.Matchers.anything;
+import static org.hamcrest.Matchers.not;
+import static org.hamcrest.core.Is.is;
+
+import android.content.Intent;
+import androidx.test.espresso.intent.rule.IntentsTestRule;
+import androidx.test.internal.runner.junit4.AndroidJUnit4ClassRunner;
+import ch.epfl.qedit.util.LocaleHelper;
+import ch.epfl.qedit.view.LoginActivity;
+import java.util.Locale;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(AndroidJUnit4ClassRunner.class)
+public class LoginActivityLanguageTest {
+    @Rule
+    public final IntentsTestRule<LoginActivity> testRule =
+            new IntentsTestRule<>(LoginActivity.class, false, false);
+
+    @Before
+    public void launchActivity() {
+        Intent intent = new Intent();
+        testRule.launchActivity(intent);
+    }
+
+    @After
+    public void finishActivity() {
+        testRule.finishActivity();
+    }
+
+    private void testChangeLanguage(
+            String language,
+            String languageCode,
+            String loginString,
+            String languageChangedString) {
+        String startupLanguage = Locale.getDefault().getLanguage();
+        // We don't use this variable but the call increases the coverage ;)
+        @SuppressWarnings("unused")
+        String lang = LocaleHelper.getLanguage(testRule.getActivity());
+
+        // Language position
+        int pos = -1;
```

Comment:
How do we know that "en" has position 0 and "fr" has position 1 ? Are the languages sorted lexicographically ? I am asking that question because I don't want the test to fail because the list gets reordered.

Reply:
You are right, we must not reorder the language list, else it will not work anymore. I tried to compute the position of the language with <the activity rule>.getActivity().getArray(R.array.languages_list) but it does not work (cannot resolve method getArray(int)